hwdb.d/60-keyboard.hwdb

```diff
@@ -91,6 +92,17 @@
 # Examples of such devices: Chromebooks where the top row is used for both
 # media and F1-F10 keys.
 
+# ######################### WAKEUP KEYS ######################################
+#
+# The wakeup keys specify which keys unblank the screen of an idle device. By
```

Comment:
So, a comment here: "wakeup" has different connotations which means a generically-named hwdb here is ambiguous.

As I mentioned above (https://github.com/systemd/systemd/pull/32261#issuecomment-2071374613) we can `EVIOCSMASK` those key events and userspace won't wake up at all for those events. That was my first interpretation but that's not what you want, right? You want userspace to still get those events but for the compositor to (not) do a very specific thing (unblanking the screen). That behaviour seems to be quite specific to the use-case once we get past volume keys.

I'm assuming volume control should still work though so some key routing to other applications is still intended so this is really just about the screen.

Which makes me wonder if this is the right place to put it? This seems a lot like specific policy (which is just fine to have) that may not be the right fit for system's default hw databases? This looks like something best shipped by whoever wants to implement this behaviour (and later shared if there's multiple parties).

Reply:
> You want userspace to still get those events but for the compositor to (not) do a very specific thing (unblanking the screen).

That's the current use case we're after. The initial version used `CHASSIS_KEYS` to convey position information rather than policy. Maybe that is the better approach?

 > That was my first interpretation but that's not what you want, right? You want userspace to still get those events but for the compositor to (not) do a very specific thing (unblanking the screen). That behaviour seems to be quite specific to the use-case once we get past volume keys.

There's other buttons on phones besides the volume keys which initially prompted this: https://gitlab.gnome.org/World/Phosh/phoc/-/issues/363 . 

> This looks like something best shipped by whoever wants to implement this behaviour (and later shared if there's multiple parties).

Hmm...so we'd install a custom hwdb entry in `/lib/udev/hwdb.d/` (reusing `60-evdev.rules`) and propose that later on upstream ? That could work to gather more use cases and than come up with more data later.

We'd basically need #32362 for that to work for DT devices too.

Reply:
> Hmm...so we'd install a custom hwdb entry in /lib/udev/hwdb.d/ (reusing 60-evdev.rules) and propose that later on upstream ? That could work to gather more use cases and than come up with more data later.

yes, there's no requirement to ship the hwdb in systemd, the hwdb entries have a 'contract' with a udev rule (which composes the lookup string) and decides when to query the hwdb but that's it. There are plenty of projects that install their own hwdb files + matching udev rule, e.g. `65-libwacom.rules` and the corresponding `65-libwacom.hwdb`.

No need to put into `60-evdev.rules`, you can pick your own prefix and separate it that way.

Reply:
I think we could go with that initially and then come back with more data. Would be great to have #32362 for that.

Reply:
yeah, I think that's the best idea for now. This way you can privately iterate over the various requirements without having to lock yourself in for a specific property format.

Happy to help where needed, just cc me.

Reply:
@whot all right. Does landing #32362 make sense from your PoV?